cfme/containers/node.py

```diff
@@ -37,63 +108,145 @@ def get_detail(self, *ident):
         return InfoBlock.text(*ident)
 
 
-@navigator.register(Node, 'All')
-class All(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+class NodeDetailsView(NodeView):
+    download = Button(name='download_view')
 
-    def am_i_here(self):
-        return match_page(summary='Nodes')
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            match_page(summary='{} (Summary)'.format(self.context['object'].name))
+        )
 
-    def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Containers', 'Container Nodes')
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
 
-    def resetter(self):
-        # Reset view and selection
-        tb.select("List View")
-        sel.check(paginator.check_all())
-        sel.uncheck(paginator.check_all())
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
 
 
 @navigator.register(Node, 'Details')
 class Details(CFMENavigateStep):
-    prerequisite = NavigateToSibling('All')
-
-    def am_i_here(self):
-        return match_page(summary='{} (Summary)'.format(self.obj.name))
+    VIEW = NodeDetailsView
+    prerequisite = NavigateToAttribute('collection', 'All')
 
     def step(self):
         # Assuming default list view from prerequisite
-        list_tbl.click_row_by_cells({'Name': self.obj.name, 'Provider': self.obj.provider.name})
+        num_pages = self.parent_view.paginator.pages_amount
+
+        # Some page logic here
+        # widgetastic paginator doesn't have iterator for pages
+        while self.parent_view.paginator.cur_page <= num_pages:
+            row = self.parent_view.nodes.row(name=self.obj.name, provider=self.obj.provider.name)
+            if row:
+                row.click()
+                return
+            # If not on this page and not on the last page, advance page and continue
+            elif self.parent_view.paginator.cur_page < num_pages:
+                # Aren't through all the pages yet
+                logger.info('Node not found, advancing page')
+                self.parent_view.paginator.next_page()
+                continue
+            # Last page reached, node never found
+            elif self.parent_view.paginator.cur_page == num_pages:
+                logger.error('Failed to locate node "{}" on provider "{}"'
+                             .format(self.obj.name, self.obj.provider.name))
+                raise NodeNotFound('Failed to navigate to node')
```

Comment:
You're right, and we've got a bunch of exceptions in cfme.exceptions that don't follow this convention. I'm going to leave it as-is for now, it would probably be good to convert all of them at once using rope if we decide its worth following the convention for them all.